bpf/aya-bpf/src/lib.rs

```diff
@@ -1,3 +1,27 @@
+//! A library to write eBPF programs.
+//!
+//! eBPF is a technology that allows running user-supplied programs inside the
+//! Linux kernel. For more info see
+//! [https://ebpf.io/what-is-ebpf](https://ebpf.io/what-is-ebpf).
+//!
+//! Aya is an eBPF library built with a focus on operability and developer
+//! experience. It does not rely on [libbpf](https://github.com/libbpf/libbpf)
+//! nor [bcc](https://github.com/iovisor/bcc) - it's built from the ground up
+//! purely in Rust, where the Rust compiler is used to build the eBPF object
+//! file. With BTF support, it offers a true
+//! [compile once, run everywhere solution](https://facebookmicrosites.github.io/bpf/blog/2020/02/19/bpf-portability-and-co-re.html),
```

Comment:
This link is solid, I did some scouring and was really surprised to find that https://docs.kernel.org basically has nothing on CO-RE except for a couple mentions in the FAQ. Not something we need to figure out for this PR, but perhaps this is an opportunity for us: if we add a really solid CO-RE section to our own documentation we can probably capture a lot of traffic from interested eBPF developers who are just looking for a "landing page" for learning about what CO-RE is and how it actually works under the hood.

I've created https://github.com/aya-rs/book/issues/79 as a follow-up for this.